src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -31,6 +33,12 @@ public IEnumerator<TElement> GetEnumerator()
                 {
                     yield return buffer._items[map[i]];
                 }
+
+                var rentedMap = Interlocked.Exchange(ref map, null);
```

Comment:
No. I'm afraid this will either be worth the cost of the interlocked, or not. (Or worse, worth it above a certain size, and not below, leaving us wondering about what sizes are seen in actual use).